### PR TITLE
Use the currently active page name in Series macro instead of subpage name

### LINKED ISSUE #379: Series macro can't be used in subpages
Using the [Series](https://github.com/gollum/gollum/wiki/Standard-Macros#series) macro in subpages looks like it tries to match the name of the subpage to the prefix instead of the current page, then fails giving an error: `Macro Error for Series: This page's name does not match the prefix '<prefix>'`.

I've had a look and will open a PR but I've never used Ruby before so might need a bit of nudging in the direction of best practice.

### PR DESCRIPTION
**Pull request by tpoliaw**
Fixes #379. I've not used Ruby before so happy to rewrite if there's a better way of doing it or a better place for the change.

### CODE REVIEW

**Comment by dometto** [comment]
@tpoliaw many thanks for spotting the bug and for this PR! The direction of this solution is fine, but we have a minor comment. We think the concept of an "active page" is not a very clear when implemented under the `Page` class: Pages are active in the frontend, but in the backend it can be confusing. So we'd like to propose that instead, you define an `active_page` method in the `Series` class:

```ruby
def active_page
  @page.parent_page&.name || @page.name
end
```

Does that seem reasonable?

**Comment by tpoliaw** [comment]
Sure, that makes sense. Is it something that would be useful for other macros as well or just series?

One comment on your suggested function, would it be better to have either `active_page` that returns the page or `active_name` that returns the name instead of a mix?

**Comment by dometto** [comment]
`active_page` (so returning page, not name) could potentially be useful for other Macros, so you could add that to the Macro class I guess, and then use `active_page.name` in the `Series` Macro! Thanks for your work!

File: lib/gollum-lib/macro.rb [hunk 1]
```diff
@@ -25,6 +25,10 @@ def html_error(s)
       "<p class=\"gollum-error\">#{s}</p>"
     end
 
+    def active_page()
```
Comment (dometto) [inline lib/gollum-lib/macro.rb:28]: just for stylistic unity, would you mind removing the parantheses here? Ruby doesn't need them :) After that I'll merge this PR!

**Comment by dometto** [comment]
Thanks @tpoliaw, and sorry for the long wait before merging!

**Pull request merged**